### PR TITLE
Full data set and validation

### PR DESCRIPTION
**Pull request by bruskiza**

### CODE REVIEW

**Comment by smn** [comment]
Please let me know when this is ready for review and I'll happily review it.

**Comment by bruskiza** [comment]
@smn ready for review... finally.

**Comment by bruskiza** [comment]
@smn really ready for review now ;)

**Comment by bruskiza** [comment]
@smn: poink.

File: src/app.js [hunk 1]
```diff
@@ -19,22 +20,41 @@ go.app = function() {
...
       var question = $("Welcome! To report a malaria case, please enter your " +
-                       "facility code. For example, 543456");
+        "facility code. For example, 543456");
       var error = $("The facility code is invalid. Please enter again.");
       return new FreeText(name, {
         question: question,
         check: function(content) {
-          var facility_codes = self.im.config.facility_codes;
-          if (facility_codes.indexOf(parseInt(content, 10)) < 0) {
+          var facility = _.result(_.find(self.im.config.facilities, {'FacCode': content}), 'Facility');
```
Comment (smn) [inline src/app.js:28]: Stashing this in `im.config` results in the full 400K of JSON being read for every single interaction, regardless of whether it's needed or not. This comes with a performance hit. We need to move it into it's own config blog (as per @justinvdm's `big_config` example he posted) or into an API.

**Comment by smn** [comment]
:+1: on the code and please merge this but the config stuff needs to be split out, that can happen in a separate PR.

**Comment by bruskiza** [comment]
au danke.

https://github.com/praekelt/go-jsbox-malaria24/issues/37 for the config split out.

**Pull request merged**